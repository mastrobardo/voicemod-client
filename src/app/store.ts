import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { soundsApi } from '../features/MainView/main-view.service';
import soundDetailsSliceReducer from '../features/SoundDetails/sound-details.slice';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {
    [soundsApi.reducerPath]: soundsApi.reducer,
    soundsDetails: soundDetailsSliceReducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat([soundsApi.middleware, logger]),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
