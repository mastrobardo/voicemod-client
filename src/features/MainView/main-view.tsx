import React, { useEffect } from "react";
import { soundsApi } from "./main-view.service";
import { Sound } from "./main-view.types";
import { Box, Grid } from "@mui/material";
import { MediaCard } from "../Card/Card";
import { NoSound } from "../NoSounds/no-sounds";
import { SoundDetails } from "../SoundDetails/sound-details";
import { useAppSelector } from "../../app/hooks";

export const MainView: React.FC = () => {
    const result = soundsApi.endpoints.getSounds.useQuery();
    const hasItems = result && result.data?.length;
    const modalId = useAppSelector((state: any) => state.soundsDetails.modalId);

    useEffect(() => {
        console.log(modalId)
    }, [modalId])

    if(!hasItems) {
        return <NoSound />
    }

    return (
        <Box sx={{ p: 2 }} >
        <SoundDetails id={modalId} open={Boolean(modalId)} />
        <Grid container spacing={3} alignItems="center" justifyContent="center">    
            {hasItems && result?.data?.map((ele:Sound, idx: number) => {
                return (
                    <Grid key={ele._id} item xs={12} sm={6} lg={3}>
                        <MediaCard {...ele} />
                    </Grid>
                )
            })}  
        </Grid>
        </Box>
    )
}