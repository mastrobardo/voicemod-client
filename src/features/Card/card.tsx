import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import { Sound } from '../MainView/main-view.types';
import { useAppDispatch } from '../../app/hooks';
import { setModalId } from '../SoundDetails/sound-details.slice';

export const MediaCard: React.FC<Sound> = ({ _id, name, icon }) => {
  const dispatch = useAppDispatch();
  const id = _id;
  const onClickHandler = (id: string) => {
    return dispatch(setModalId({id}))
  }

  return (
    <Card data-id={_id}>
      <CardMedia
        component="img"
        height="140"
        image={icon}
        alt={name}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {name}
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small" onClick={() => onClickHandler(id)}> Open Details</Button>
      </CardActions>
    </Card>
  );
}