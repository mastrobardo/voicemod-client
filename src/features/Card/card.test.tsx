import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Sound } from '../MainView/main-view.types';
import { store } from '../../app/store';

import { MediaCard } from './card';
describe('<MediaCard>', () => {
    const props: Sound = {
        _id: '123',
        icon: 'anIcon',
        name: 'aName',
        playbacks: 0,
        price: 0
    }
    const initilState = {}    

    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Provider store={store}><MediaCard {...props} /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
})
