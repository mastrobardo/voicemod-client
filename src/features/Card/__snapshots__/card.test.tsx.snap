// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCard> renders correctly when there are no items 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-id="123"
>
  <img
    alt="aName"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
    height="140"
    src="anIcon"
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      aName
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
       Open Details
    </button>
  </div>
</div>
`;
